healthbridgeltd/nodejs-sam-bootstrap
# Add Webpack support

## Overview

Adding webpack support to use single package.json for multiple lambda functions.
Add uglify plugin to reduce the code footprint, which improves performance and cost.
This will also make it easy to add typescript support.

## Problem

After upgrading to NodeJS 12, libraries loaded in included internal packages didn't work.


## Change log


- Add Webpack cli
- Add aws-sam-webpack-plugin
- Add terser-webpack-plugin for uglifying code
- Remove npm recursive install as not needed.


## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected). This requires a major version bump.

## How can this be tested?

- Checkout the repo
- Run locally using
```
make run
```
- deploy to your sandbox using
```
make deploy
```
The endpoints should work as expected
## Checklist
